pennylane/beta/tapes/tape.py

```diff
@@ -1103,7 +1103,7 @@ def numeric_pd(self, idx, device, params=None, **options):
         order = options.get("order", 1)
         h = options.get("h", 1e-7)
 
-        shift = np.zeros_like(params)
+        shift = np.zeros_like(params, dtype=np.float64)
```

Comment:
A bug I noticed while writing the tests. If `params` is an integer array, than `np.zeros_like(params)` will also be an integer array, resulting in `shift[idx] = 1e-7` being rounded down to 0.